R/get_rcode.R

```diff
@@ -59,106 +56,40 @@
 #' @references [show_rcode_modal()], [get_rcode_header()]
 get_rcode <- function(datasets = NULL,
                       datanames = `if`(is.null(datasets), datasets, datasets$datanames()),
-                      merge_expression = "",
                       chunks = teal.code::get_chunks_object(),
                       selected_chunk_ids = character(0),
-                      session = getDefaultReactiveDomain(),
+                      session = NULL,
                       title = NULL,
                       description = NULL) {
   checkmate::assert_class(datasets, "FilteredData", null.ok = TRUE)
-  checkmate::assert_string(merge_expression)
-  if (merge_expression != "") {
-    message(paste0(
-      "'merge_expression' argument of 'get_rcode()' will be deprecated.",
-      " Please use 'chunks_push_data_merge()'."
-    ))
-  }
   if (!inherits(chunks, "chunks")) {
     stop("No code chunks given")
   }
   checkmate::assert_string(title, null.ok = TRUE)
   checkmate::assert_string(description, null.ok = TRUE)
-
   rlang::push_options(width = 120)
 
+  if (!is.null(session)) {
+    lifecycle::deprecate_warn("0.11.2", "get_rcode(session)")
+  }
+
```

Comment:
So there's a specific lifecycle argument for deprecating just an argument

![image](https://user-images.githubusercontent.com/15201933/176469435-44721455-d788-412b-adbc-9a9fef0985ac.png)